### ISSUE TITLE
Bug: Mobile error cursor jumps to start line

### ISSUE THREAD

**Issue by reymon359** [facebook/lexical#4207]
Hey there 👋🏽  

When I click in the end of a word/line, the cursor jumps to the beginning of it

Here is an example with the lexical playground

https://user-images.githubusercontent.com/31936665/227817927-b3d3ec6c-2b49-4341-adf3-1cce9481ac85.mp4

Lexical version:  locally version `"lexical": "0.7.6"` although it is also happening on the latest version as of the moment I am writing this 😉 

## Steps To Reproduce

1. on Mobile write some text
2. then click like if you want to continue writing in the end on a line
3. you will be jumped to the beginning of the line

Link to code example: https://playground.lexical.dev/

## The current behavior

When I click in the end of a word/line, the cursor jumps to the beginning of it

## The expected behavior

When I click in the end of a word/line, the cursor should be in the end of the word/line

**Comment by reymon359** [comment]
Hey @zurfyx any update on this or a way to "quick" fix it while it is properly addressed? 😄

**Comment by reymon359** [comment]
I repinged here https://discord.com/channels/953974421008293909/953974421486436393/1099820714669899859

**Comment by reymon359** [comment]
I had a look to the playground, and it is still happening :(

https://user-images.githubusercontent.com/31936665/234759182-59161ba2-7c45-4218-a000-5c41ae179f53.mp4

**Comment by dragonman225** [comment]
Yeah, it's happening on my phone, too. I'm using Google Chrome 116.0.5845.172 on Android 13.


https://github.com/facebook/lexical/assets/25148955/be3d77c9-7840-4ec7-8883-536cf73361eb

**Comment by dragonman225** [comment]
Still happening on Google Chrome 126.0.6478.71 Android 14.


https://github.com/facebook/lexical/assets/25148955/00ad4b0b-edbb-4b54-8727-d579ab0a5e87
